src/main/groovy/wooga/gradle/build/unity/tasks/AbstractUnityBuildEngineTask.groovy

```diff
@@ -0,0 +1,161 @@
+package wooga.gradle.build.unity.tasks
+
+
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFile
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import wooga.gradle.build.unity.internal.BuildEngineArgs
+import wooga.gradle.secrets.internal.Secrets
+import wooga.gradle.unity.UnityTask
+
+import javax.crypto.spec.SecretKeySpec
+
+
+abstract class AbstractUnityBuildEngineTask extends UnityTask {
+
+    private final Property<String> build
+    private final Property<String> version
+    private final Property<String> config
+    private final Property<String> exportMethodName
+    private final Property<String> outputPath
+    private final RegularFileProperty secretsFile
+    private final Property<SecretKeySpec> secretsKey
+    private final ListProperty<Object> customArguments
+
+    AbstractUnityBuildEngineTask() {
+        this.build = project.objects.property(String)
+        this.version = project.objects.property(String)
+        this.config = project.objects.property(String)
+        this.exportMethodName = project.objects.property(String)
+        this.outputPath = project.objects.property(String)
+        this.secretsFile = project.objects.fileProperty()
+        this.secretsKey = project.objects.property(SecretKeySpec.class)
+        this.customArguments = project.objects.listProperty(Object)
+    }
+
+    BuildEngineArgs defaultArgs() {
+        Provider<Directory> outputDir = outputPath.map { path ->
+            project.layout.projectDirectory.dir(path)
+        }
+        def secrets = secretsFile.map { RegularFile secretsFile ->
+            Secrets.decode(secretsFile.asFile.text)
+        }
+        def environmentSecrets = project.provider({
+            generateSecretsEnvironment(secrets, secretsKey)
+        }.memoize())
+
+        def buildEngineArgs = new BuildEngineArgs(project.providers, exportMethodName)
+        buildEngineArgs.with {
+            addArg("--build", build)
+            addArg("--version", version)

```

Comment:
Since this property is mandatory only for the player build request, I would make it optional.

Reply:
Removed this arg, as it isn't needed for basic builds.